# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/SourceSurfaceWebgl.h
# Commit: f30dd1bddb47
# Full Hash: f30dd1bddb471a4f3487fe0e55e8c4bad0f8962a
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-02-12 09:47:43
# Regressor Bug: 1754130
# File Overlap Count: 2
# Description:
#   Bug 1754130 - Support presenting a WebGLFramebuffer to its own swap chain without opaque FB. r=aosmond,jgilbert
#   
#   Most of the support for presenting a WebGLFramebuffer to a swap chain existed as part of the
#   mechanism for opaque WebXR framebuffer support. However, such "opaque" framebuffer are meant
#   to be opaque in the sense that their attachments can't be inspected or changed, which does
# ==============================================================================

diff -r e178aa7d969e -r f30dd1bddb47 dom/canvas/SourceSurfaceWebgl.h
--- a/dom/canvas/SourceSurfaceWebgl.h	Fri Feb 11 19:49:55 2022 +0000
+++ b/dom/canvas/SourceSurfaceWebgl.h	Fri Feb 11 19:49:56 2022 +0000
@@ -10,8 +10,7 @@
 #include "mozilla/gfx/2D.h"
 #include "mozilla/WeakPtr.h"
 
-namespace mozilla {
-namespace gfx {
+namespace mozilla::gfx {
 
 class DrawTargetWebgl;
 class TextureHandle;
@@ -38,6 +37,8 @@
   bool Map(MapType aType, MappedSurface* aMappedSurface) override;
   void Unmap() override;
 
+  bool HasReadData() const { return !!mData; }
+
  private:
   friend class DrawTargetWebgl;
 
@@ -63,7 +64,6 @@
   RefPtr<TextureHandle> mHandle;
 };
 
-}  // namespace gfx
-}  // namespace mozilla
+}  // namespace mozilla::gfx
 
 #endif /* MOZILLA_GFX_SOURCESURFACEWEBGL_H_ */